### PR TITLE
Minmize distro API

### PR DESCRIPTION
**Pull request by pellared**
## Why

> One or more configuration variables may be needed to properly configure GDI repositories. Components that can be configured with environment variables MUST support configuration of these variables using environment variables. Any component that cannot be configured with environment variables MUST support configuration of these variables using an alternate method. Any component MAY support configuration of these variables by additional methods.

Supporting only a single configuration method is simplifying the codebase and adoption.

From: https://github.com/signalfx/gdi-specification/blob/main/specification/configuration.md

## What

- Minimize `github.com/signalfx/splunk-otel-go/distro` API to
  contain only necessary option functions.
  - Remove `WithAccessToken` function,
    use `SPLUNK_ACCESS_TOKEN` environment variable instead.
  - Remove `WithEndpoint` function,
    use one of the
    `OTEL_EXPORTER_OTLP_ENDPOINT`, `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT`, `OTEL_EXPORTER_JAEGER_ENDPOINT`
    environment variables instead.
  - Remove `WithPropagator` function,
    use `OTEL_PROPAGATORS` environment variable instead.
  - Remove `WithTraceExporter` function,
    use `OTEL_TRACES_EXPORTER` environment variable instead.
- Remove most of `distro/README.md` as I find our [official doc](https://docs.splunk.com/Observability/gdi/get-data-in/application/go/configuration/advanced-go-otel-configuration.html#nav-Configure-the-Go-instrumentation) in better shape. I plan to create a `docs/advanced-config.go` based on our [official doc](https://docs.splunk.com/Observability/gdi/get-data-in/application/go/configuration/advanced-go-otel-configuration.html#nav-Configure-the-Go-instrumentation) in the following days. It should be easier to keep both of them in sync that way and the structure will be similar to our other distros.

## Testing 

| Scenario | Trace |
| - | - |
| default config | ✅  |
| `SPLUNK_REALM=us0` | ✅|
| `OTEL_TRACES_EXPORTER=jaeger-thrift-splunk SPLUNK_REALM=us0` | ✅|
| `OTEL_TRACES_EXPORTER=jaeger-thrift-splunk OTEL_EXPORTER_JAEGER_ENDPOINT=http://localhost:14268/api/traces` | ✅|

### CODE REVIEW

File: distro/exporter.go [hunk 1]
```diff
@@ -44,67 +43,45 @@ var exporters = map[string]traceExporterFunc{
...
 		opts = append(opts, otlptracegrpc.WithHeaders(map[string]string{
 			"X-Sf-Token": c.AccessToken,
 		}))
 	}
 
+	if c.TLSConfig != nil {
+		tlsCreds := credentials.NewTLS(c.TLSConfig)
+		opts = append(opts, otlptracegrpc.WithTLSCredentials(tlsCreds))
+	} else if noneEnvVarSet(otelExporterOTLPEndpointKey, otelExporterOTLPTracesEndpointKey, splunkRealmKey) {
+		// we assume that the default endpoint (local collector) is non-TLS
```
Comment (MrAlias) [inline distro/exporter.go:61]:
```suggestion
		// Assume that the default endpoint (local collector) is non-TLS.
```

File: distro/exporter.go [hunk 2]
```diff
@@ -44,67 +43,45 @@ var exporters = map[string]traceExporterFunc{
...
+func noneEnvVarSet(envs ...string) bool {
+	for _, env := range envs {
+		if _, ok := os.LookupEnv(env); ok {
+			return false
+		}
 	}
+	return true
+}
 
+// otlpEndpoint returns the endpoint to use for the OTLP gRPC exporter
```
Comment (MrAlias) [inline distro/exporter.go:78]:
```suggestion
// otlpEndpoint returns the endpoint to use for the OTLP gRPC exporter.
```

**Review by MrAlias — APPROVED** [review]
LGTM. Minor docs concerns only.

File: CHANGELOG.md [hunk 3]
```diff
@@ -31,6 +31,19 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
+  - Remove `WithAccessToken` function,
+    use `SPLUNK_ACCESS_TOKEN` environment variable instead.
+  - Remove `WithEndpoint` function,
+    use one of the
+    `OTEL_EXPORTER_OTLP_ENDPOINT`, `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT`, `OTEL_EXPORTER_JAEGER_ENDPOINT`
+    environment variables instead.
+  - Remove `WithPropagator` function,
+    use `OTEL_PROPAGATORS` environment variable instead.
+  - Remove `WithTraceExporter` function,
+    use `OTEL_TRACES_EXPORTER` environment variable instead.
```
[the remark this answers is not in the document]
Comment (pellared) [reply CHANGELOG.md:48]: Addressed https://github.com/signalfx/splunk-otel-go/pull/941/commits/82c063f688ded777de8d065d4be8019e80f54870

**Pull request merged**